lib/srv/app/aws/endpoints.go

```diff
@@ -65,6 +66,16 @@ func resolveEndpoint(r *http.Request) (*endpoints.ResolvedEndpoint, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	// Use X-Forwarded-Host header if it is a valid AWS endpoint.
+	forwardedHost := r.Header.Get("X-Forwarded-Host")
+	if awsapiutils.IsAWSEndpoint(forwardedHost) {
```

Comment:
Right now the IsAWSEndpoint checks for `.amazonaws.com"` suffixes but we should still supporting AWS CN where the endpoint is set to *.com.cn`

Reply:
`IsAWSEndpoint` checks if uri **Contains** `.amazonaws.com`, which tests true for AWS CN endpoints. I guess `IsAWSEndpoint` is not the best implementation where we didn't trim port then test suffix.

Reply:
So `IsAWSEndpoint` will return true for `foo.amazonaws.com.teleport.sh`, right?

Reply:
yes, have to add schema tho if not there. will keep it like that for now.